Gordon360/Services/MembershipService.cs

```diff
@@ -88,167 +92,108 @@ public MEMBERSHIP GetSpecificMembership(int membershipID)
 
         /// <summary>
         /// Fetches all the membership information linked to the student whose id appears as a parameter.
         /// </summary>
         /// <param name="username">The student's AD Username.</param>
         /// <returns>A MembershipViewModel IEnumerable. If nothing is found, an empty IEnumerable is returned.</returns>
-        public async Task<IEnumerable<MembershipViewModel>> GetMembershipsForStudentAsync(string username)
+        public IEnumerable<MembershipView> GetMembershipsByUser(string username)
         {
-            var account = _context.ACCOUNT.FirstOrDefault(x => x.AD_Username.Trim() == username);
+            var account = _context.ACCOUNT.FirstOrDefault(a => a.AD_Username.Trim() == username);
             if (account == null)
             {
                 throw new ResourceNotFoundException() { ExceptionMessage = "The Account was not found." };
             }
 
-            var memberships = await _context.Procedures.MEMBERSHIPS_PER_STUDENT_IDAsync(int.Parse(account.gordon_id));
-
-            return memberships.OrderByDescending(x => x.StartDate).Select(m => new MembershipViewModel
-            {
-                MembershipID = m.MembershipID,
-                ActivityCode = m.ActivityCode.Trim(),
-                ActivityDescription = m.ActivityDescription.Trim(),
-                ActivityImagePath = m.ActivityImagePath,
-                SessionCode = m.SessionCode.Trim(),
-                SessionDescription = m.SessionDescription.Trim(),
-                IDNumber = m.IDNumber,
-                FirstName = m.FirstName.Trim(),
-                LastName = m.LastName.Trim(),
-                Participation = m.Participation.Trim(),
-                ParticipationDescription = m.ParticipationDescription.Trim(),
-                StartDate = m.StartDate,
-                EndDate = m.EndDate,
-                Description = m.Description,
-                ActivityType = m.ActivityType.Trim(),
-                ActivityTypeDescription = m.ActivityTypeDescription.Trim(),
-                GroupAdmin = m.GroupAdmin,
-                Privacy = m.Privacy,
-            });
+            return _context.MembershipView.Where(m => m.Username == account.AD_Username).OrderByDescending(x => x.StartDate);
         }
 
         /// <summary>
         /// Fetches the number of followers associated with the activity whose code is specified by the parameter.
         /// </summary>
         /// <param name="activityCode">The activity code.</param>
         /// <returns>int.</returns>
-        public async Task<int> GetActivityFollowersCountAsync(string activityCode)
+        public int GetActivityFollowersCount(string activityCode)
         {
-            var memberships = await GetMembershipsForActivityAsync(activityCode);
-            return memberships.Where(x => x.Participation == "GUEST").Count();
+            return _context.MembershipView.Where(m => m.ActivityCode == activityCode && m.Participation == "GUEST").Count();
```

Comment:
Can we extract the "GUEST" string to a constant, like the other roles?

Reply:
Also, won't this get the count of memberships for the activity across all sessions? I don't think that's the intended behavior.

Instead, I would think this can be combined with the `GetActivitySubscribersCountForSession` method below, taking a nullable sessionCode that defaults to the current session, right?